### PR TITLE
Cache settings

### PR DESCRIPTION
**Pull request by tehrengruber**
This PR adds the ability to pass the cache directory (both `root_path` and `dir_name`) to the `stencil` decorator. In FVM we compile multiple stencils in parallel and by default use a common cache path project-wide regardless of where the driver resides. Setting this directory in our stencil wrapper lead to a race-condition which we can avoid with the changes here.

### CODE REVIEW

**Comment by tehrengruber** [comment]
bors try

File: src/gt4py/gtscript.py [hunk 1]
```diff
@@ -235,6 +243,8 @@ def stencil(
         raise ValueError(f"Invalid 'name' string ('{name}')")
     if not isinstance(rebuild, bool):
         raise ValueError(f"Invalid 'rebuild' bool value ('{rebuild}')")
+    if cache_settings is not None and not isinstance(cache_settings, dict):
+        raise ValueError(f"Invalid 'cache_settings' dictionary ('{rebuild}')")
```
Comment (DropD) [inline src/gt4py/gtscript.py:247]: copy pasta? 
```suggestion
        raise ValueError(f"Invalid 'cache_settings' dictionary ('{cache_settings}')")
```

File: src/gt4py/gtscript.py [hunk 2]
```diff
@@ -260,13 +270,17 @@ def stencil(
...
+        cache_settings = {}
+
     build_options = gt_definitions.BuildOptions(
         name=name,
         module=module,
         format_source=format_source,
         rebuild=rebuild,
         backend_opts=kwargs,
         build_info=build_info,
+        cache_settings=cache_settings,
```
Comment (DropD) [inline src/gt4py/gtscript.py:283]: Have you considered inlining the if block above?
```suggestion
        cache_settings=cache_settings or {},
```

**Review by DropD — CHANGES_REQUESTED** [review]
👍 There is really just this one typo...

File: src/gt4py/gtscript.py (hunk 1 above)
Comment (tehrengruber) [reply src/gt4py/gtscript.py:247]: wups

**Review by DropD — APPROVED** [review]

**Pull request merged**